src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -4736,6 +4738,13 @@ CalcHeatBalanceOutsideSurf( Optional_int_const ZoneToResimulate ) // if passed i
 		HGrdExtSurf( SurfNum ) = 0.0;
 		QRadLWOutSrdSurfs( SurfNum ) = 0.0;
 
+		// Calculate heat extract due to additional heat flux source term as the surface boundary condition 
+		if( Surface( SurfNum ).HasAdditionalHeatSourceTerm ) {
```

Comment:
With my comment above, this double layer if block would reduce to just a single `if HasOutsideHeatSourceTerm` block.